Mathlib/RingTheory/TwoSidedIdeal/Lattice.lean

```diff
@@ -0,0 +1,135 @@
+/-
+Copyright (c) 2024 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+
+import Mathlib.RingTheory.TwoSidedIdeal.Basic
+
+/-!
+# The complete lattice structure on two-sided ideals
+-/
+
+namespace TwoSidedIdeal
+
+variable (R : Type*) [NonUnitalNonAssocRing R]
+
+instance : Sup (TwoSidedIdeal R) where
+  sup I J := { ringCon := I.ringCon ⊔ J.ringCon }
+
+lemma sup_ringCon (I J : TwoSidedIdeal R) : (I ⊔ J).ringCon = I.ringCon ⊔ J.ringCon := rfl
```

Comment:
I am not sure if these should be tagged with `simp`. Ideally, when people use `TwoSidedIdeal`, they shouldn't be seeing `TwoSidedIdeal.ringCon` all the time. What do you think @eric-wieser?

Reply:
I agree that people shouldn't really see this, but if they do, I think having this as a `simp` lemma is fine.